server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -718,7 +704,7 @@ private ExternalReaderManager createReaderManager(RefreshWarmerListener external
     private DirectoryReader getDirectoryReader() throws IOException {
         // for segment replication: replicas should create the reader from store, we don't want an open IW on replicas.
```

Comment:
Lets update this comment with a bit more on why we need to wrap our reader with a SoftDeletesDirectoryReaderWrapper.